src/odemis/driver/symphotime.py

```diff
@@ -0,0 +1,888 @@
+
+            # convert the OptionalDataRecord type to a regular python dictionary for logging
+            data = {r.name : r.data for r in decoded_msg.odOptional.values()}
+            logging.debug("Acq: %s", data)
+
+            # update metadata and vigilant attributes based on data coming back
+            # from the server
+            if "ServerVersion" in data:
+                self._metadata[model.MD_SW_VERSION] = data['ServerVersion']
+            if "ResultingFilename" in data:
+                self.scanner.filename.value = data['ResultingFilename']
+            if "ResultingGroupname" in data:
+                self.scanner.directory.value = data['ResultingGroupname']
+
+        elif isinstance(decoded_msg, EncodedStatusReplyMessage):
+            # Do not answer these types of messages
+            # Log the updated status from the server
+            err_status = decoded_msg.ecStatus
+            logging.debug("Status: 0x%x: %s", err_status, ERRCODE[err_status])
+
+            if err_status == PQ_STOPREASON_CODE_FINISHED_OK:
+                if self.isMeasuring():
+                    logging.info("Finished ok.")
+                    self._measurinement_stopped.set()
+            elif err_status < 0:
+                self._measurinement_stopped.set()
+                raise SPTError(err_status)
+
+        elif isinstance(decoded_msg, DataframeServerReplyMessage):
+            # Do not answer these types of messages
+            # Log the updated status from the server
+            err_status = decoded_msg.ecStatus
+            logging.debug("Status Reply: 0x%x: %s", err_status, ERRCODE[err_status])
+
+            if err_status == PQ_STOPREASON_CODE_CONTINUE_OK:
+                if self.isMeasuring() and self._user_break:
+                    logging.info("Server acknowledged user break.")
+                    self._measurinement_stopped.set()
+                    self._user_break = False
+            elif err_status < 0:  # an error
+                self._measurinement_stopped.set()
+                raise SPTError(err_status)
+
+        else:
+            logging.warning("Unknown message received. Msgtype %s", type(decoded_msg))
+
+
+class Scanner(model.Emitter):
+    '''
+    A scanner that is a child of the Symphotime parent. Acts as a wrapper for the Symphotime server
+    VA's:
+        filename (string) with extension *.ptu
+        directory (string)
+        resolution(resolution (int, int))
+        bidirectional (bool)
+        dwellTime (float)
+    '''
+    def __init__(self, name, role, parent, **kwargs):
+        '''
+        parent (symphotime.Controller): a symphotime server parent object
+        '''
+        # we will fill the set of children with Components later in ._children
+        model.Emitter.__init__(self, name, role, parent=parent, **kwargs)
+        
+        # Define VA's as references to the parent. 
+        self.filename = model.StringVA(setter=self._updateFilename)
```

Comment:
Normally, we use `_updateXXX` for a subscriber, and `_setXXX` for a setter. Could you rename the `_updateFilename` and `_updateDirectory` to `_setFilename` and `_setDirectory`?